### PR TITLE
[DOC][SPIR-V] Add Joint Matrix to composite types

### PR DESCRIPTION
**Pull request by MrSidims**
And reuse OpConstantComposite to create a matrix.

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by AlexeySotkin** [comment]
I believe you should update the `Version` and `Revision history` tables too.

File: sycl/doc/extensions/SPIRV/SPV_INTEL_joint_matrix.asciidoc [hunk 1]
```diff
@@ -53,7 +53,7 @@ please let us know!
 [width="40%",cols="25,25"]
 |========================================
 | Last Modified Date | {docdate}
```
Comment (AlexeySotkin) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_joint_matrix.asciidoc:55]: Please set the current date

File: sycl/doc/extensions/SPIRV/SPV_INTEL_joint_matrix.asciidoc [hunk 2]
```diff
@@ -477,4 +492,5 @@ Revision History
 |Rev|Date|Author|Changes
 |1|2021-02-16|Alexey Sotkin|Initial revision
 |1|2021-09-06|Dmitry Sidorov|Split OpJointMatrixMadINTEL instruction into 4
+|1|2021-12-28|Dmitry Sidorov|Add Joint Matrix to Composite definition
```
Comment (AlexeySotkin) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_joint_matrix.asciidoc:495]: Why it is still "1" and why in the Revision table the revision is "B"?

Comment (MrSidims) [reply sycl/doc/extensions/SPIRV/SPV_INTEL_joint_matrix.asciidoc:495]: > why in the Revision table the revision is "B"?

Well, it was "A" there :)

**Review by AlexeySotkin — APPROVED** [review]

**Pull request merged**